tests/test_spidermiddlewares.py

```diff
@@ -367,3 +367,24 @@ def test_read_data_middleware():
 
     assert len(transformed_items) == 1
     assert transformed_items[0]['data'] == b'{}'
+
+
+@pytest.mark.parametrize('exception', [BadZipFile(), Exception()])
+def test_retry_data_error_middleware(exception):
+    spider = spider_with_crawler()
+    response = response_fixture()
+    middleware = RetryDataErrorMiddleware()
+
+    generator = middleware.process_spider_exception(response, exception, spider)
+
+    if isinstance(exception, BadZipFile):
+        request = next(generator)
+        assert request.dont_filter
+        assert request.meta['retries'] == 1
+        assert request.url == response.request.url
+        response.request.meta['retries'] = 3
+        generator = middleware.process_spider_exception(response, exception, spider)
+        assert not list(generator)
+
```

Comment:
Use whitespace to visually distinguish assertions from setup code.

```suggestion

        assert request.dont_filter is True
        assert request.meta['retries'] == 1
        assert request.url == response.request.url

        response.request.meta['retries'] = 3
        generator = middleware.process_spider_exception(response, exception, spider)

        assert not list(generator)
```